src/services/document_service.py

```diff
@@ -306,7 +305,7 @@ def _get_node_to_update(self, reference: str, node_entity: Union[dict, list]) ->
 
     def update_document(
         self,
-        reference: str,
+        address: Address,
```

Comment:
could update docstring:

"What to update is referred to with a reference string." -> "What to update is referred to with an address."